manifests/init.pp

```diff
@@ -420,6 +424,11 @@
     validate_string($ssl_protocol)
   }
 
+  if $yum_max_speed {
+    validate_string($yum_max_speed)
+    to_bytes($yum_max_speed)
```

Comment:
I don't think this works, $yum_max_speed will still contain a string.  You should do something like

```
$real_yum_max_speed = to_bytes($yum_max_speed)
```

And then use $real_yum_max_speed in the template